onert-micro/eval-driver/TrainingDriver.cpp

```diff
@@ -100,117 +92,236 @@ int entry(int argc, char **argv)
+                       i * sizeof(MODEL_TYPE) * INPUT_SIZE);
+
+      readDataFromFile(input_target_test_data_path, reinterpret_cast<char *>(test_target),
+                       sizeof(float) * OUTPUT_SIZE * cur_batch_size,
+                       i * sizeof(MODEL_TYPE) * OUTPUT_SIZE);
+
+      train_interpreter.setInput(reinterpret_cast<uint8_t *>(test_input), 0);
+      train_interpreter.setTarget(reinterpret_cast<uint8_t *>(test_target), 0);
+
+      float mse = 0.f;
+      float mae = 0.f;
+      float cross_entropy_metric = 0.f;
+      float accuracy = 0.f;
+
+      train_interpreter.evaluateMetric(
+        onert_micro::MAE_METRICS, reinterpret_cast<void *>(&cross_entropy_metric), cur_batch_size);
+      train_interpreter.evaluateMetric(onert_micro::MSE_METRICS,
+                                       reinterpret_cast<void *>(&accuracy), cur_batch_size);
+
+      accuracy_v.push_back(accuracy);
+      cross_entropy_v.push_back(cross_entropy_metric);
     }
-    printf("\n");
-    test_data_u8 += size;
+    sum_acc = std::accumulate(accuracy_v.begin(), accuracy_v.end(), 0.f);
+    sum_ent = std::accumulate(cross_entropy_v.begin(), cross_entropy_v.end(), 0.f);
+    std::cout << "Test Average MSE = " << sum_acc / accuracy_v.size() << "\n";
+    std::cout << "Test Average MAE = " << sum_ent / cross_entropy_v.size() << "\n";
   }
 
-  float mse_result = 0.0f;
-
-  settings.metric = luci_interpreter::training::MSE;
-  onert_micro_training.test(num_train_data_samples, reinterpret_cast<const uint8_t *>(train_data),
-                            reinterpret_cast<const uint8_t *>(label_train_data),
-                            reinterpret_cast<void *>(&mse_result));
-
-  float mae_result = 0.0f;
-
-  settings.metric = luci_interpreter::training::MAE;
-  onert_micro_training.test(num_train_data_samples, reinterpret_cast<const uint8_t *>(train_data),
-                            reinterpret_cast<const uint8_t *>(label_train_data),
-                            reinterpret_cast<void *>(&mae_result));
-
-  printf("MSE_ERROR TRAIN = %f\n", mse_result);
-
-  printf("MAE_ERROR TRAIN = %f\n", mae_result);
-
-  mse_result = 0.0f;
-
-  settings.metric = luci_interpreter::training::MSE;
-  onert_micro_training.test(num_test_data_samples, reinterpret_cast<const uint8_t *>(test_data),
-                            reinterpret_cast<const uint8_t *>(label_test_data),
-                            reinterpret_cast<void *>(&mse_result));
-
-  mae_result = 0.0f;
-
-  settings.metric = luci_interpreter::training::MAE;
-  onert_micro_training.test(num_test_data_samples, reinterpret_cast<const uint8_t *>(test_data),
-                            reinterpret_cast<const uint8_t *>(label_test_data),
-                            reinterpret_cast<void *>(&mae_result));
-
-  printf("MSE_ERROR TEST = %f\n", mse_result);
-
-  printf("MAE_ERROR TEST = %f\n", mae_result);
+  // Save training result
+  saveModel(output_trained_file_path, circle_model);
```

Comment:
This may be exposed via API. Thus, circle_model or model_ptr should be kept in some context.

Reply:
So, move this function insert OMTrainingInterpreter, right?